packages/@lwc/synthetic-shadow/src/faux-shadow/node.ts

```diff
@@ -173,7 +173,11 @@ function cloneNodePatched(this: Node, deep?: boolean): Node {
 function childNodesGetterPatched(this: Node): NodeListOf<Node> {
     if (isSyntheticShadowHost(this)) {
         const owner = getNodeOwner(this);
-        const childNodes = isNull(owner) ? [] : getAllMatches(owner, getFilteredChildNodes(this));
+        const filteredChildNodes = getFilteredChildNodes(this);
+        // no need to filter nodes by owner in case of light DOM
```

Comment:
This also applies to root shadow DOM components:
```suggestion
        // No need to filter by owner for non-shadowed nodes
```